--- a/debian/libssl3t64.symbols
+++ b/debian/libssl3t64.symbols 
@@ -9,6 +9,9 @@
  *@OPENSSL_3.3.0 3.3.0
  *@OPENSSL_3.4.0 3.4.0
  *@OPENSSL_3.5.0 3.5.0
+ OPENSSL_3.0.1@OPENSSL_3.0.1 3.5.5-1gl10
+ ossl_ctx_legacy_digest_signatures_allowed@OPENSSL_3.0.1 3.5.5-1gl10
+ ossl_ctx_legacy_digest_signatures_allowed_set@OPENSSL_3.0.1 3.5.5-1gl10
 libssl.so.3 libssl3t64 #MINVER#
 * Build-Depends-Package: libssl-dev
  *@OPENSSL_3.0.0 3.0.0
