:100644 100644 b3b95ed76b 0000000000 M	test/recipes/30-test_evp_data/evpmd_sha.txt

diff --git a/test/recipes/30-test_evp_data/evpmd_sha.txt b/test/recipes/30-test_evp_data/evpmd_sha.txt
index b3b95ed76b..b3d4da9411 100644
--- a/test/recipes/30-test_evp_data/evpmd_sha.txt
+++ b/test/recipes/30-test_evp_data/evpmd_sha.txt
@@ -14,17 +14,17 @@
 
-Title = SHA tests from (RFC6234 section 8.5 and others)
-
-Digest = SHA1
-Input = "abc"
-Output = a9993e364706816aba3e25717850c26c9cd0d89d
-
-Digest = SHA1
-Input = "abcdbcdecdefdefgefghfghighijhijkijkljklmklmnlmnomnopnopq"
-Output = 84983e441c3bd26ebaae4aa1f95129e5e54670f1
-
-Digest = SHA1
-Input = "a"
-Ncopy = 1000
-Count = 1000
-Output = 34aa973cd4c4daa4f61eeb2bdbad27316534016f
+#Title = SHA tests from (RFC6234 section 8.5 and others)
+#
+#Digest = SHA1
+#Input = "abc"
+#Output = a9993e364706816aba3e25717850c26c9cd0d89d
+#
+#Digest = SHA1
+#Input = "abcdbcdecdefdefgefghfghighijhijkijkljklmklmnlmnomnopnopq"
+#Output = 84983e441c3bd26ebaae4aa1f95129e5e54670f1
+#
+#Digest = SHA1
+#Input = "a"
+#Ncopy = 1000
+#Count = 1000
+#Output = 34aa973cd4c4daa4f61eeb2bdbad27316534016f
 
:100644 100644 d14005a89a 0000000000 M	doc/man7/OSSL_PROVIDER-FIPS.pod

diff --git a/doc/man7/OSSL_PROVIDER-FIPS.pod b/doc/man7/OSSL_PROVIDER-FIPS.pod
index d14005a89a..688b1b464d 100644
--- a/doc/man7/OSSL_PROVIDER-FIPS.pod
+++ b/doc/man7/OSSL_PROVIDER-FIPS.pod
@@ -429,4 +429,2 @@ Symmetric cipher tests used with the "KAT_Cipher" type.
 
-=item "SHA1" (B<OSSL_SELF_TEST_DESC_MD_SHA1>)
-
 =item "SHA2" (B<OSSL_SELF_TEST_DESC_MD_SHA2>)
:100644 100644 de736794a9 0000000000 M	include/openssl/self_test.h

diff --git a/include/openssl/self_test.h b/include/openssl/self_test.h
index de736794a9..4161985b4b 100644
--- a/include/openssl/self_test.h
+++ b/include/openssl/self_test.h
@@ -63,3 +63,2 @@ extern "C" {
 #define OSSL_SELF_TEST_DESC_ASYM_RSA_DEC "RSA_Decrypt"
-#define OSSL_SELF_TEST_DESC_MD_SHA1 "SHA1"
 #define OSSL_SELF_TEST_DESC_MD_SHA2 "SHA2"
:100644 100644 6abab0a7a1 0000000000 M	providers/fips/self_test_data.inc

:100644 100644 6abab0a7a1 0000000000 M	providers/fips/self_test_data.inc

diff --git a/providers/fips/self_test_data.inc b/providers/fips/self_test_data.inc
index 6abab0a7a1..14a3a0e2b9 100644
--- a/providers/fips/self_test_data.inc
+++ b/providers/fips/self_test_data.inc
@@ -200,8 +200,2 @@ static const ST_KAT_DIGEST st_kat_digest_tests[] =
 {
-    {
-         OSSL_SELF_TEST_DESC_MD_SHA1,
-         "SHA1",
-         ITM_STR(sha1_pt),
-         ITM(sha1_digest),
-    },
     {
